src/MongoDB.Driver.Core/Core/WireProtocol/Messages/CommandMessageSection.cs

```diff
@@ -137,12 +139,15 @@ public Type0CommandMessageSection(TDocument document, IBsonSerializer<TDocument>
     /// <summary>
     /// Represents a Type 1 CommandMessage section.
     /// </summary>
-    public class Type1CommandMessageSection : CommandMessageSection
+    public abstract class Type1CommandMessageSection : CommandMessageSection
     {
         // private fields
         private readonly IBatchableSource<object> _documents;
         private readonly IBsonSerializer _documentSerializer;
+        private readonly IElementNameValidator _elementNameValidator;
         private readonly string _identifier;
+        private readonly int? _maxBatchCount;
+        private readonly int? _maxDocumentSize;
```

Comment:
Turns out we needed a bit more information about the payload, hence the new fields.